src/main/java/com/weiliang/task/Deadline.java

```diff
@@ -0,0 +1,26 @@
+package com.weiliang.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    private LocalDateTime timing;
+
+    public Deadline(String task, String timing) {
+        super(task);
+        this.timing = LocalDateTime.parse(timing);
+    }
+
+    @Override
+    public String toFormattedString() {
+        return "D | " + (completed ? 1 : 0) + " | " + task + " | " + timing;
+    }
+
+    @Override
+    public String toString() {
+        return "[D][" + (completed ? "X" : " ") + "] " + task + " (by: "
+                + timing.format(DateTimeFormatter.ofPattern("dd MMMM YYYY, hh:mma")) + ")";
+    }
+
+}
```

Comment:
Follows Java Coding Standards which is good. A suggestion would be to add header comments to methods being override to improve readability and ease of understanding especially when tenary operator is used which could make it slightly less readable.